app/api.go

```diff
@@ -18,13 +18,18 @@ type BuildPreProcessor interface {
 }
 
 // CreateApp creates an application from the specified json application descriptor
-func CreateApp(env env.Project, appJson string) error {
+func CreateApp(env env.Project, appJson string, appName string) error {
 
 	descriptor, err := ParseAppDescriptor(appJson)
 	if err != nil {
 		return err
 	}
 
+	if appName != "" {
+		appJson = strings.Replace(appJson, `"` + descriptor.Name + `"`, `"` + appName + `"`, 1)
```

Comment:
I was parsing the json at first, but the output was horrible... the default go marshaller was putting fields in arbitrary places...  I'll work on making a custom marshaller shortly